src/libs/actions/Workflow.ts

```diff
@@ -0,0 +1,224 @@
+import type {OnyxCollection, OnyxUpdate} from 'react-native-onyx';
+import Onyx from 'react-native-onyx';
+import * as API from '@libs/API';
+import type {CreateWorkspaceApprovalParams, RemoveWorkspaceApprovalParams, UpdateWorkspaceApprovalParams} from '@libs/API/parameters';
+import {WRITE_COMMANDS} from '@libs/API/types';
+import {convertApprovalWorkflowToPolicyEmployees} from '@libs/WorkflowUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {ApprovalWorkflow, Policy} from '@src/types/onyx';
+import type {Approver, Member} from '@src/types/onyx/ApprovalWorkflow';
+
+let currentApprovalWorkflow: ApprovalWorkflow | undefined;
+Onyx.connect({
+    key: ONYXKEYS.APPROVAL_WORKFLOW,
+    callback: (approvalWorkflow) => {
+        currentApprovalWorkflow = approvalWorkflow;
+    },
+});
+
+let allPolicies: OnyxCollection<Policy>;
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.POLICY,
+    waitForCollectionCallback: true,
+    callback: (value) => (allPolicies = value),
+});
+
+let authToken: string | undefined;
+Onyx.connect({
+    key: ONYXKEYS.SESSION,
+    callback: (value) => {
+        authToken = value?.authToken;
+    },
+});
+
+function createApprovalWorkflow(policyID: string, approvalWorkflow: ApprovalWorkflow) {
+    const policy = allPolicies?.[policyID];
+
+    if (!authToken || !policy) {
+        return;
+    }
+
+    const previousEmployeeList = {...policy.employeeList};
+    const previousApprovalMode = policy.approvalMode;
+    const updatedEmployees = convertApprovalWorkflowToPolicyEmployees({approvalWorkflow, employeeList: previousEmployeeList});
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.APPROVAL_WORKFLOW,
+            value: {
+                isLoading: true,
+            },
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                employeeList: updatedEmployees,
+                approvalMode: CONST.POLICY.APPROVAL_MODE.ADVANCED,
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.APPROVAL_WORKFLOW,
+            value: {...approvalWorkflow, isLoading: false},
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                employeeList: previousEmployeeList,
+                approvalMode: previousApprovalMode,
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.SET,
+            key: ONYXKEYS.APPROVAL_WORKFLOW,
+            value: null,
+        },
+    ];
```

Comment:
Clearing this workflow should have been done in the `optimisticData` because the data is no longer needed as soon as the form is submitted.

Also clearing this in `successData` caused a bug if you submit the form offline and go back to the workflow page and get online. The page will suddenly becomes blank (https://github.com/Expensify/App/issues/51468).